### PR TITLE
Ensure that we error when calling "const extern fn" with wrong convention

### PR DESCRIPTION
**Pull request by Aaron1011**
See #64926

### CODE REVIEW

**Comment by RalfJung** [comment]
(I hope this is not urgent, I am traveling this week)

**Comment by Aaron1011** [comment]
This is just a test to ensure that the current, correct behavior doesn't change. It's not urgent at all.

File: src/test/ui/consts/const-extern-fn/abi-mismatch.rs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// Checks that we report ABI mismatches for "const extern fn"
+// compile-flags: -Z unleash-the-miri-inside-of-you
```
Comment (RalfJung) [inline src/test/ui/consts/const-extern-fn/abi-mismatch.rs]: We usually have all miri-unleash tests in the corresponding folder, so I think we should do the same here... Cc @oli-obk

**Comment by Aaron1011** [comment]
@RalfJung: Updated

**Comment by RalfJung** [comment]
Thanks!

@bors r+ rollup=always

**Pull request merged**